gsi/gridmap_eppn_callout/globus_gridmap_eppn.c

```diff
@@ -615,24 +616,45 @@ globus_gridmap_eppn_callout(
         context, subject, &found_identity, shared_user_cert, shared_user_chain, shared_exp);
     if(result == GLOBUS_SUCCESS)
     {
-        if(desired_identity && strcmp(found_identity, desired_identity) != 0)
+        if(desired_identity == NULL)
         {
-            GLOBUS_GRIDMAP_CALLOUT_ERROR(
-                result,
-                GLOBUS_GRIDMAP_CALLOUT_LOOKUP_FAILED,
-                ("Credentials specify id of %s, can not allow id of %s.\n",
-                 found_identity, desired_identity));
-            globus_free(found_identity);
-            goto error;
+            if(getenv("GLOBUS_EPPN_FORCE_GRIDMAP"))
+            {
+                result = globus_gss_assist_gridmap(found_identity, &tmp_identity);
+                if(result != 0)
+                {
+                    GLOBUS_GRIDMAP_CALLOUT_ERROR(
+                        result,
+                        GLOBUS_GRIDMAP_CALLOUT_LOOKUP_FAILED,
+                        ("Could not map eppn %s (subject %s)\n", found_identity, subject));
+                    globus_free(found_identity);
+                    goto error;
+                }
+                globus_free(found_identity);
+                found_identity = tmp_identity;
+            }
+        }
+        else
+        {
+            if(strcmp(found_identity, desired_identity) != 0)
+            {
+                GLOBUS_GRIDMAP_CALLOUT_ERROR(
+                    result,
+                    GLOBUS_GRIDMAP_CALLOUT_LOOKUP_FAILED,
+                    ("Credentials specify id of %s, can not allow id of %s.\n",
+                     found_identity, desired_identity));
+                globus_free(found_identity);
+                goto error;
+            }
         }
     }
     else
     {
         /* proceed with gridmap lookup */
         if(desired_identity == NULL)
         {
-            rc = globus_gss_assist_gridmap(subject, &found_identity);
-            if(rc != 0)
+            result = globus_gss_assist_gridmap(subject, &found_identity);
+            if(result != 0)
```

Comment:
globus_gss_assist_gridmap returns int, so "rc" as before this change is correct. "result" is globus_result_t and doesn't match.